src/method/query/response.rs

```diff
@@ -0,0 +1,237 @@
+
+    fn deref(&self) -> &Self::Target {
+        &self.0[..]
+    }
+}
+
+impl DerefMut for QueryResponse {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0[..]
+    }
+}
+
+impl IntoIterator for QueryResponse {
+    type Item = QueryResult;
+    type IntoIter = <Vec<QueryResult> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl Index<usize> for QueryResponse {
+    type Output = QueryResult;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.0[index]
+    }
+}
+
+impl AsRef<Vec<QueryResult>> for QueryResponse {
+    fn as_ref(&self) -> &Vec<QueryResult> {
+        &self.0
+    }
+}
+
+/// A wrapper type around the result of a single query.
+///
+/// Provides utility functions to deserialize items (of type [Value]) into any
+/// [`<T>`] that implements [DeserializeOwned].
+#[derive(Debug, Clone)]
+pub struct QueryResult(Result<Vec<Value>>);
+
+impl QueryResult {
+    /// Returns the deserialized [`<T>`] from the inner [Value]s over the given
+    /// range or index.
+    ///
+    /// - if `index_or_range` is an index of type [usize] then a single [`Option<T>`]
+    /// is returned.
+    /// - if `index_or_range` is a range then a [`Option<Vec<T>>`] is returned if
+    /// and only if the full range is found inside the inner list.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # #[derive(Debug, serde::Deserialize)]
+    /// # #[allow(dead_code)]
+    /// # struct Account {
+    /// #   id: String,
+    /// #   balance: String
+    /// # }
+    /// #
+    /// # use surrealdb_rs::{Result, Surreal};
+    /// # use surrealdb_rs::net::WsClient;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// # let client = Surreal::<WsClient>::new();
+    /// # let token = String::new();
+    /// let response = client.query("select * from account where balance = '$100'").await?;
+    ///
+    /// if let Some(first_query_result) = response.get_query(0) {
+    ///   // print the first account:
+    ///   let account: Option<Account> = first_query_result.get(0)?;
+    ///   dbg!(account);
+    ///
+    ///   // print the first two accounts, if at least two accounts are in the response:
+    ///   let accounts: Option<Vec<Account>> = first_query_result.get(0..2)?;
+    ///   dbg!(accounts);
+    /// }
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn get<T, I>(&self, index_or_range: I) -> Result<Option<T>>
+    where
+        T: DeserializeOwned,
+        I: SliceIndex<[Value]>,
+        <I as SliceIndex<[surrealdb::sql::Value]>>::Output: Serialize,
+    {
+        let values: &Vec<Value> = self.0.as_ref().map_err(|error| error.clone())?.as_ref();
+
+        let some_slice = values.get::<I>(index_or_range);
+        let items = match some_slice {
+            Some(slice) => {
+                let bytes = serde_pack::to_vec(slice)?;
+                let response = serde_pack::from_slice(&bytes)?;
```

Comment:
This is what `from_value` does. Let's use that instead.

Reply:
This code was a little different than the `from_value` since it worked on a slice. I've created an internal function `from_serializable` that accepts anything serializable, and which is now used here & by `from_value` as well.
commit: [b91b9dc](https://github.com/surrealdb/surrealdb.rs/pull/11/commits/b91b9dc9d256d5c341ce06126ba73b80fcfabcf5)